lighthouse-core/test/report/v2/report-renderer-test.js

```diff
@@ -0,0 +1,177 @@
+
+  before(setupJsDomGlobals);
+  after(cleanupJsDomGlobals);
+
+  describe('setText', () => {
+    it('sets text of element', () => {
+      const el = document.createElement('div');
+      const result = DOM.setText(el, 'hello world');
+      assert.deepStrictEqual(result, el);
+      assert.equal(result.textContent, 'hello world');
+    });
+
+    it('does not fail when given null element argument', () => {
+      assert.doesNotThrow(() => {
+        const result = DOM.setText(null, 'hello world');
+        assert.strictEqual(result, null);
+      });
+    });
+  });
+
+  describe('addClass', () => {
+    it('adds classes to element', () => {
+      const el = document.createElement('div');
+      const result = DOM.addClass(el, 'class1', 'class2');
+      assert.ok(result.classList.contains, 'class1');
+      assert.ok(result.classList.contains, 'class2');
+    });
+
+    it('does not fail when given null element argument', () => {
+      assert.doesNotThrow(() => {
+        const result = DOM.addClass(null, 'class1');
+        assert.strictEqual(result, null);
+      });
+    });
+  });
+
+  describe('createElement', () => {
+    it('creates a simple element using default values', () => {
+      const dom = new DOM(document);
+      const el = dom.createElement('div');
+      assert.equal(el.localName, 'div');
+      assert.equal(el.className, '');
+      assert.equal(el.className, el.attributes.length);
+    });
+
+    it('creates an element from parameters', () => {
+      const dom = new DOM(document);
+      const el = dom.createElement('div', 'class1 class2', {title: 'title attr', tabindex: 0});
+      assert.equal(el.localName, 'div');
+      assert.equal(el.className, 'class1 class2');
+      assert.equal(el.getAttribute('title'), 'title attr');
+      assert.equal(el.getAttribute('tabindex'), '0');
+    });
+  });
+
+  describe('cloneTemplate', () => {
+    it('should clone a template', () => {
+      const dom = new DOM(document);
+      const clone = dom.cloneTemplate('#tmpl-lighthouse-audit-score');
+      assert.ok(clone.querySelector('.lighthouse-score'));
+    });
+
+    it('fails when template cannot be found', () => {
+      const dom = new DOM(document);
+      assert.throws(() => {
+        const clone = dom.cloneTemplate('#unknown-selector'); // eslint-disable-line no-unused-vars
+      });
+    });
+  });
+});
+
+describe('ReportRenderer V2', () => {
+  const window = self;
+
+  before(setupJsDomGlobals);
+  after(cleanupJsDomGlobals);
+
+  describe('renderReport', () => {
+    const ReportRenderer = window.ReportRenderer;
+
+    it('should render a report', () => {
+      const renderer = new ReportRenderer(document);
+      const output = renderer.renderReport(sampleResults);
+      assert.ok(output.classList.contains('lighthouse-report'));
+    });
+
+    it('should render an exception for invalid input', () => {
+      const renderer = new ReportRenderer(document);
+      const output = renderer.renderReport({});
```

Comment:
Since we're not really validating anything right now this relies on implementation details. What about something explicit like
```js
const output = renderer.renderReport({
  get reportCategories() { throw new Error() }
});
```